browser-test/src/northstar_program_deep_link.test.ts

```diff
@@ -36,91 +36,100 @@ test.describe('navigating to a deep link', {tag: ['@northstar']}, () => {
     await logout(page)
   })
 
-  test('shows a login prompt for guest users', async ({page}) => {
-    await page.goto('/programs/test-deep-link')
-    await expect(
-      page
-        .getByRole('dialog', {name: 'Create an account or sign in'})
-        .getByRole('heading'),
-    ).toBeVisible()
-    await validateAccessibility(page)
-  })
-
-  test('does not show login prompt for logged in users', async ({page}) => {
-    await loginAsTestUser(page)
-    await page.goto('/programs/test-deep-link')
-    await expect(
-      page.getByRole('dialog', {name: 'Create an account or sign in'}),
-    ).toHaveCount(0)
-  })
+  test.describe('after starting an application', () => {
+    test.beforeEach(async ({applicantQuestions}) => {
+      await applicantQuestions.applyProgram(programName, true)
+      await applicantQuestions.answerTextQuestion('first answer')
+      await applicantQuestions.clickContinue()
+    })
+    test('shows a login prompt for guest users', async ({page}) => {
+      await page.goto('/programs/test-deep-link')
+      await expect(
+        page
+          .getByRole('dialog', {name: 'Create an account or sign in'})
+          .getByRole('heading'),
+      ).toBeVisible()
+      await validateAccessibility(page)
+    })
 
-  test('takes guests and logged in users through the flow correctly', async ({
-    page,
-    applicantQuestions,
-  }) => {
-    // Exercise guest path
-    // Act
-    await page.goto('/programs/test-deep-link')
-    await page.getByRole('link', {name: 'Continue to application'}).click()
+    test('does not show login prompt for logged in users', async ({page}) => {
+      await loginAsTestUser(page)
+      await page.goto('/programs/test-deep-link')
+      await expect(
+        page.getByRole('dialog', {name: 'Create an account or sign in'}),
+      ).toHaveCount(0)
+    })
 
-    // Assert
-    await page.getByRole('link', {name: 'Continue'}).click()
-    await applicantQuestions.validateQuestionIsOnPage(questionText)
+    test('takes guests and logged in users through the flow correctly', async ({
+      page,
+      applicantQuestions,
+    }) => {
+      // Exercise guest path
+      await page.goto('/programs/test-deep-link')
+      // Sees the login prompt
+      await page.getByRole('link', {name: 'Continue to application'}).click()
+      // Is taken to the review page
+      await expect(
```

Comment:
**suggestion:** Consider bringing this out into a helper function in applicant questions, since we use it multiple times throughout this test.

Reply:
Done.